File: https:/github.com/rajesh-kayal-dev/Leetcode.git/75-SortColors/75-SortColors.js
// Last updated: 6/15/2026, 10:02:39 AM
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var sortColors = function(nums) {

    let i = 0;
    let j = 0;
    let k = nums.length -1;
    
    const swap = (nums, i , j)=>{
            let temp = nums[i];
            nums[i] = nums[j];
            nums[j] = temp;
    } 

    while(i <= k){
        if(nums[i] === 0){
            swap(nums, i,j);
            i++;
            j++;
        }else if(nums[i] === 2){
            swap(nums, i, k);
            k--;
        }else{
            i++;
        }
    }
    

};

File: https:/github.com/rajesh-kayal-dev/Leetcode.git/75-SortColors/75-SortColors.java
// Last updated: 6/15/2026, 10:02:46 AM
class Solution {
    public void sortColors(int[] nums) {
        int n =nums.length;
        int noOfZeros=0,noOfOnes=0;
        for(int i =0;i<n;i++){
            if (nums[i]==0) noOfZeros++;
            if (nums[i]==1) noOfOnes++;
        }
        for (int i =0;i<n;i++){
            if(i<noOfZeros)
            nums[i]=0;
            else if(i<noOfZeros+noOfOnes)
            nums[i]=1;
            else
            nums[i]=2;
        }
    }
}